fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabsStrip.kt

```diff
@@ -0,0 +1,369 @@
+                    val selectedItemInfo =
+                        listState.layoutInfo.visibleItemsInfo.firstOrNull { it.key == selectedTab.id }
+
+                    if (listState.isItemPartiallyVisible(selectedItemInfo) || selectedItemInfo == null) {
+                        listState.animateScrollToItem(state.tabs.indexOf(selectedTab))
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun LazyListState.isItemPartiallyVisible(itemInfo: LazyListItemInfo?) =
+    itemInfo != null &&
+        (itemInfo.offset + itemInfo.size > layoutInfo.viewportEndOffset || itemInfo.offset < 0)
+
+@Composable
+private fun TabItem(
+    state: TabStripItem,
+    modifier: Modifier = Modifier,
+    onCloseTabClick: (id: String) -> Unit,
+    onSelectedTabClick: (id: String) -> Unit,
+) {
+    TabStripCard(
+        modifier = modifier.fillMaxSize(),
+        backgroundColor =
+        if (state.isPrivate) {
+            if (state.isSelected) {
+                FirefoxTheme.colors.layer3
+            } else {
+                FirefoxTheme.colors.layer2
+            }
+        } else {
+            if (state.isSelected) {
+                FirefoxTheme.colors.layer2
+            } else {
+                FirefoxTheme.colors.layer3
+            }
+        },
+        elevation = if (state.isSelected) {
+            selectedTabStripCardElevation
+        } else {
+            defaultTabStripCardElevation
+        },
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxSize()
+                .clickable { onSelectedTabClick(state.id) },
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
+            Row(
+                modifier = Modifier.weight(1f, fill = false),
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                Spacer(modifier = Modifier.size(8.dp))
+
+                TabStripIcon(state.url)
+
+                Spacer(modifier = Modifier.size(8.dp))
+
+                Text(
+                    text = state.title,
+                    color = FirefoxTheme.colors.textPrimary,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    style = FirefoxTheme.typography.subtitle2,
+                )
+            }
+
+            IconButton(onClick = { onCloseTabClick(state.id) }) {
+                Icon(
+                    painter = painterResource(R.drawable.mozac_ic_cross_20),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                    contentDescription = stringResource(R.string.close_tab),
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun TabStripIcon(url: String) {
+    Box(
+        modifier = Modifier
+            .size(tabStripIconSize)
+            .clip(CircleShape),
+        contentAlignment = Alignment.Center,
+    ) {
+        Favicon(
```

Comment:
I hope my last comment with this block wasn't misleading. We should still use the `icon` bitmap from `TabSessionState.content` here, if it's not null. Otherwise, `Favicon` will be doing a network call or making a creating custom asset for _every_ tab icon.

Reply:
https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/TabThumbnail.kt#74-91

Reply:
I did do that but then realised that Favicon does cache icons in memory, similar to how `ContentState.icon` is in memory, so we don't have to worry about it making multiple requests. Or am I misunderstanding something here?

Reply:
OK so I had to double check. Favicon does indeed pull from the same memory cache that drives `ContentState.icon` (via `BrowserIcons`). 

I suppose it would come down to some theoretical trade offs of:
- Using the bitmap that's already out of cache and inside of `ContentState.icon`
- Is additional loading from memory within `BrowserIcons` trivial? [source code](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt#291-297)

I don't have the scientific answers to those, but if the bitmap is already inside of `ContentState.icon`, we shouldn't have to pull it out of the cache again.

Reply:
I think it's fine in this case as the cache (LRUCache) is simply holding the bitmap, and the logic is simply getting the Bitmap, there's no decoding from bytes happening. And It simplifies the UI even further.

Reply:
OK, I'm fine with moving forward with this.